### PR TITLE
Issue 3436

### PR DESCRIPTION
**Pull request by himasagaratluri**
## Problem Statement

What is the problem you're trying to solve?
The problem is deeply explained in the linked issue, the change is relative simple, adding validation to make sure we evaluate StringType secret for the aws secretsmanager store

## Related Issue
https://github.com/external-secrets/external-secrets/issues/3436
Fixes #...

## Proposed Changes

How do you like to solve the issue and why?

Add utility to compare the values of string type and update the clause to validate string type

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -507,3 +507,18 @@ func dig[T any](key string, data map[string]any) (t T, _ error) {
...
+	if len(stringToByteSlice) != len(valueByte) {
+		return false
+	}
+
+	for sb := range valueByte {
+		if stringToByteSlice[sb] != valueByte[sb] {
+			return false
+		}
+	}
+
```
Comment (gusfcarvalho) [inline pkg/utils/utils.go:522]: this function would currently break users with `secretBinary` set (as you can see with the unit tests failures). I think an if is needed here:
```suggestion

func CompareStringAndByteSlices(valueString *string, valueByte []byte) bool {
        if valueString == nil {
          return false
         }
	stringToByteSlice := []byte(*valueString)
	if len(stringToByteSlice) != len(valueByte) {
		return false
	}

	for sb := range valueByte {
		if stringToByteSlice[sb] != valueByte[sb] {
			return false
		}
	}
```
You can check if this worked by running the unit tests locally with `make test`

Comment (himasagaratluri) [reply pkg/utils/utils.go:522]: Good catch! wonder how my local tests passed. Fix is on the way.

**Review by gusfcarvalho — APPROVED** [review]

**Comment by gusfcarvalho** [comment]
thanks for your contribution @himasagaratluri !!

**Pull request merged**